internal/backends/collection.go

```diff
@@ -98,6 +103,9 @@ type QueryResult struct {
 // The passed context should be used for canceling the initial query.
 // It also can be used to close the returned iterator and free underlying resources,
 // but doing so is not necessary - the handler will do that anyway.
+//
+// If SortPushdown is set to false, any sort, or limit pushdown won't be performed,
+// even when Sort, or Limit fields are set.
```

Comment:
If that's true, then we don't need that field at all – the handler should just not set `Sort` and `Limit` fields.